File: server/model/EnvironmentShare.py
from sqlalchemy import BigInteger, Column, String, ForeignKey

from database.db import Base
from sqlalchemy.dialects.postgresql import ENUM


# ! Classe associativa para relacionar outros usuários a um ambiente
class VotingUserEnvironment(Base):
    __tablename__ = "users_environments"
    __table_args__ = {"schema": "ic"}

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    user_id: str = Column(String(45), ForeignKey("ic.users.id"), nullable=False)
    environment_id: str = Column(
        String(45), ForeignKey("ic.environments.id"), nullable=False
    )
    role: str = Column(ENUM("owner", "read", name="mining_type"), nullable=False)

File: server-js/models/EnvironmentShare.js
module.exports = (sequelize, DataTypes) => {
  let EnvironmentShare = sequelize.define(
    "EnvironmentShare",
    {
      id: {
        type: DataTypes.BIGINT,
        primaryKey: true,
        autoIncrement: true,
      },
      role: {
        type: DataTypes.ENUM("owner", "read"),
        allowNull: false,
      },
    },
    {
      tableName: "environment_share",
      schema: "ic",
    }
  );

  return EnvironmentShare;
};
/*
EnvironmentShare.init(
  {
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true,
    },
    
    !! Relacionamentos - fazer dps
    user_id: {
      type: DataTypes.STRING(45),
      allowNull: false,
      references: {
        model: "users",
        key: "id",
      },
    },
    !! Relacionamentos - fazer dps
    environment_id: {
      type: DataTypes.STRING(45),
      allowNull: false,
      references: {
        model: "environments",
        key: "id",
      },
    },
    role: {
      type: DataTypes.ENUM("owner", "read"),
      allowNull: false,
    },
  },
  {
    sequelize,
    modelName: "EnvironmentShare",
    tableName: "environment_share",
    schema: "ic",
  }
);*/
